tiny_dnn/layers/layer.h

```diff
@@ -670,6 +785,33 @@ class layer : public node {
     post_update();
   }
 
+  void update_parameters(optimizer *optimizer_ptr, size_t batch_size) {
+    float_t rcp_batch_size  = float_t{1} / float_t{batch_size};
+    Tensor<float_t> &diff_t = parameters_diff_;
+    for (Parameter *parameter : parameters_) {
+      if (trainable() && parameter->is_trainable()) {
+        parameter->merge_grads(&diff_t);
+
+        for (size_t j = 0; j < diff_t.size(); ++j) {
+          diff_t.host_at(j) *= rcp_batch_size;
```

Comment:
@Randl just thinking in future now that I see a lots of `host_at`. Should we have a transparent API `at(size_t i)` which internally implements `host_at` and `device_at` ??